src/Sparrow/Json/Parsing/JsonParserState.cs

```diff
@@ -45,15 +45,19 @@ public static int VariableSizeIntSize(int value)
             return count;
         }
 
-        public static int FindEscapePositionsMaxSize(StringSegment str, out int escapedCount)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int FindEscapePositionsMaxSize(string str, out int escapedCount)
+        {
+            return FindEscapePositionsMaxSize(str.AsSpan(), out escapedCount);
+        }
+        
+        public static int FindEscapePositionsMaxSize(ReadOnlySpan<char> str, out int escapedCount)
         {
             var count = 0;
             var controlCount = 0;
 
-            for (int i = 0; i < str.Length; i++)
+            foreach (var value in str)
```

Comment:
Not sure if usage of `for` instead of `foreach` wasn't intentional because of slightly better performance. Let's get back to `for` just in case.